### ISSUE TITLE
Wrap selection

### ISSUE THREAD

**Issue by Skydancer9** [atom/snippets#317]
Hi,
Sorry for my beginner's question. Is it possible to wrap selected text in snippet code?
Example:
Select some text, for example number 10 below:
`<p>Some text<sup>10</sup></p>`

AFTER (inserts a snippet and adds the selected number 10 in three places):
`<p>Some text<sup><a epub:type="noteref" role="doc-noteref" href="#fn10" id="ref10">10</a></sup></p>`

**Comment by savetheclocktower** [comment]
No, but you can define a custom command in your `init.js` to do this.

Haven't tested this, but here's how your example could work:

```js
atom.commands.add('atom-text-editor', {
  'custom:some-command': () => {
    let editor = atom.workspace.getActiveTextEditor();
    let selection = editor.getLastSelection();

    let output = `<a epub:type="noteref" role="doc-noteref" href="#fn${selection} id="ref${selection}">${selection}</a>`;

    selection.insertText(output);
  }
});
```

Feel free to name it whatever you want, of course. Then you could map it to a keystroke in your `keymap.cson`:

```coffee
'atom-text-editor':
  'ctrl-shift-0': 'custom:some-command'
```

There's a similar programmatic interface to the snippets module, so you could define a snippet on-the-fly that incorporates the selection. But in your example, there's no reason to do so; it's simpler to do it with Atom's built-in APIs.

**Comment by Skydancer9** [comment]
@savetheclocktower Thank you!

**Comment by pbsds** [comment]
Could it be possible to add support for the `$TM_SELECTED_TEXT` [variable](https://macromates.com/manual/en/environment_variables), or `$VISUAL` as in vim?
